test/functional/reports_controller_test.rb

```diff
@@ -66,14 +62,24 @@ def create_a_report
     @report = Report.import JSON.parse(File.read(File.expand_path(File.dirname(__FILE__) + "/../fixtures/report-empty.json")))
   end
 
-  def user_setup
-    @request.session[:user] = users(:one).id
-    users(:one).roles       = [Role.find_by_name('Anonymous'), Role.find_by_name('Viewer')]
-  end
+  context 'when user does not have permission to view hosts' do
+    setup    { setup_user 'view', 'reports' }
+    teardown { assert_response :not_found }
 
-  test 'user with viewer rights should succeed in viewing reports' do
```

Comment:
Not quite sure if this should be deleted - the other test uses an admin account rather than an ordinary user, it may have a bit of value.